app/assets/javascripts/discourse/app/components/d-lightbox.js

```diff
@@ -0,0 +1,476 @@
+import {
+  ANIMATION_DURATION,
+  KEYBOARD_SHORTCUTS,
+  LAYOUT_TYPES,
+  LIGHTBOX_APP_EVENT_NAMES,
+  LIGHTBOX_ELEMENT_ID,
+  SWIPE_DIRECTIONS,
+  TITLE_ELEMENT_ID,
+} from "discourse/lib/lightbox/constants";
+import {
+  createDownloadLink,
+  getSwipeDirection,
+  openImageInNewTab,
+  preloadItemImages,
+  scrollParentToElementCenter,
+  setCarouselScrollPosition,
+  setSiteThemeColor,
+} from "discourse/lib/lightbox/helpers";
+
+import Component from "@glimmer/component";
+import { bind } from "discourse-common/utils/decorators";
+import discourseLater from "discourse-common/lib/later";
+import { htmlSafe } from "@ember/template";
+import { inject as service } from "@ember/service";
+import { tracked } from "@glimmer/tracking";
+
+export default class DLightbox extends Component {
+  @service appEvents;
+
+  @tracked callbacks = {};
+  @tracked items = [];
+  @tracked options = {};
```

Comment:
I don't need it needs to be tracked, it's only getting initial state when opening the lightbox, but not changed afterwards.

Might be the same for items and callbacks.